### PR TITLE
Experiment: add cowboy_swagger

### PR DESCRIPTION
**Pull request by vkatsuba**
Remove 9.1 MB of static files. Covering API as was in http_api_spec.yaml. Provide correct Swagger API for metrics.
---
Reasons o PR:
 * Remove all static files of Swagger
 * Erlang developers can create specs in Erlang maps without any issues
 * Easier to maintain and keep up to date

Link: http://localhost:8000/api-docs/index.html

**P.S.** This is only proposal. This PR can be rejected in any time, coz this is experiment.

### CODE REVIEW

File: src/ergw_http_api_doc.erl [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        {DefinitionResponseStatus, PropertyResponseStatus},
+        {DefinitionResponseStatusAcceptNew, PropertyResponseStatusAcceptNew}
+    ],
+    
+    %% Add definitions
+    [cowboy_swagger:add_definition(Def, Prop) || {Def, Prop} <- DefinitionsAndProperties],
+
+    % In path
+    InPathStatusAcceptNew = #{
+            name => <<"value">>,
```
Comment (vkatsuba) [inline src/ergw_http_api_doc.erl:77]: align code block into left

**Comment by RoadRunnr** [comment]
In general, 3GPP first describes the API using OpenAPI YAML, then we implement this. So, for those APIs there should be no need to add a generated documentation.

For our own APIs, it wouldn't hurt if we thought about them and documented them first in the same way.

I'm not a big fan of auto-documenting what is there (except for very mechanic and repetitive things, like the prometheus metrics).

**Comment by RoadRunnr** [comment]
There are projects that generate functions form OpenAPI specs. Maybe that would be a better way to make sure that we fullfile the promises of specs?

**Comment by vkatsuba** [comment]
> There are projects that generate functions form OpenAPI specs. Maybe that would be a better way to make sure that we fullfile the promises of specs?

Do you mean that - create logic for parse YAML of Swagger specs and try detect if dispatch have the same API as described in YAML and if YAML and API from dispatch will not be equal we will provide an some errors about this?

**Comment by RoadRunnr** [comment]
> > There are projects that generate functions form OpenAPI specs. Maybe that would be a better way to make sure that we fullfile the promises of specs?
> 
> Do you mean that - create logic for parse YAML of Swagger specs and try detect if dispatch have the same API as described in YAML and if YAML and API from dispatch will not be equal we will provide an some errors about this?

more something like this: https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator/src/main/resources/erlang-server

**Comment by vkatsuba** [comment]
> more something like this: https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator/src/main/resources/erlang-server
Got it, will be investigated, thanks.

**Comment by vkatsuba** [comment]
Closed by reason that this integration is not needed for current project.

**Pull request closed without merging**